src/api/cluster_create.go

```diff
@@ -110,9 +110,44 @@ func (a *ClusterAPI) CreateCluster(c *gin.Context) {
 			createClusterRequest.SecretId = secret.GenerateSecretIDFromName(createClusterRequest.SecretName)
 		}
 
-		commonCluster, err := a.createCluster(ctx, &createClusterRequest, orgID, userID, createClusterRequest.PostHooks)
+		existingCluster, err := a.clusterManager.GetClusterByName(ctx, orgID, createClusterRequest.Name)
 		if err != nil {
-			c.JSON(err.Code, err)
+			c.JSON(http.StatusInternalServerError, err)
```

Comment:
**Question**: shouldn't we also handle not found error here in case no cluster exists at this point? Or does it not return an error, but only a `nil, nil`?

Reply:
Thanks for bringing my attention to this part again, we shouldn't return the error here since the expected behavior when creating a cluster is that the cluster is not existing yet. I think the error should be ignored, the check whether the`existingCluster` is `nil` in the next part should be enough.